### ISSUE TITLE
Hash "#" URL Issue Caused by <base href=""> Tag in Production

### ISSUE THREAD

**Issue by KikoTs** [slavafyi/vite-plugin-shopify-import-maps#5]
It appears that the presence of the `<base href="">` tag is causing issues with relative and hash URLs, leading them to broken pages.

For instance, consider the link:
`<a href="#page">Click me</a>`
This incorrectly redirects to: `https://domain.myshopify.com/cdn/shop/t/200/assets/#page`

Similarly, issues arise with other special characters like "&" and "?".

Any insights or solutions to address this would be greatly appreciated.

**Comment by slavafyi** [comment]
@KikoTs Yeah, I figured that might be a problem. Actually the target for the base url should be the url of the script in which the module is imported. For example, if we have `theme.js`:

```html
<script type="module" src="https://domain.myshopify.com/cdn/shop/t/200/assets/theme.js?v=123456"></script>
````

Then all modules imported into `theme.js` should use the base url `https://domain.myshopify.com/cdn/shop/t/200/assets/`. But I couldn't get that to work correctly, so I added a `<base>` tag. We probably need a different solution, I'll try to figure it out today/tomorrow.
